ddl/table.go

```diff
@@ -738,12 +739,37 @@ func onUpdateFlashReplicaStatus(t *meta.Meta, job *model.Job) (ver int64, _ erro
 	if err != nil {
 		return ver, errors.Trace(err)
 	}
-	if tblInfo.TiFlashReplica == nil || (tblInfo.TiFlashReplica.Available == available) {
+	if tblInfo.TiFlashReplica == nil || (tblInfo.TiFlashReplica.Available == available) ||
+		(tblInfo.ID != physicalID && available == tblInfo.TiFlashReplica.IsPartitionAvailable(physicalID)) {
 		return ver, nil
 	}
 
-	if tblInfo.TiFlashReplica != nil {
+	if tblInfo.ID == physicalID {
 		tblInfo.TiFlashReplica.Available = available
+	} else if pi := tblInfo.GetPartitionInfo(); pi != nil {
```

Comment:
What if `pi == nil`? Should it report an error?

Reply:
great, done.